test/int/cluster_test.go

```diff
@@ -160,6 +160,22 @@ var _ = Describe("AtlasCluster", func() {
 					Expect(c.DiskSizeGB).To(BeAssignableToTypeOf(float64ptr(0)), "DiskSizeGB is no longer a *float64, please check the spec!")
 				})
 			})
+
+			By("Pausing the cluster", func() {
+				createdCluster.Spec.Paused = boolptr(true)
+				performUpdate()
+				checkAtlasState(func(c *mongodbatlas.Cluster) {
+					Expect(c.Paused).To(Equal(createdCluster.Spec.Paused))
+				})
```

Comment:
can we check that after the cluster paused/unpaused - the following updates to the other fields can be done? 
Also I guess we need to make sure that those updates don't change the cluster in Atlas (paused/unpaused) back to `nil`

Reply:
[Atlas does not support updates on paused clusters:](https://docs.atlas.mongodb.com/reference/api/clusters-modify-one/#request-body-parameters)
> You cannot modify a paused cluster other than to resume the cluster. Nor can you pause a cluster with pending changes.

Reply:
ok, thanks for this clarification.
Then I guess the following behavior is expected?
- the user updates "paused: true"
- then the user changes other fields in the spec, controller sends this to Atlas but no changes happen there as the cluster is paused? The AtlasCluster resource is in "ready: true" state though?
- then the user updates "pause: false" and the changes done in the previous step reach Atlas now?

WDYT about testing the above scenario or maybe you'll find a better scenario to check

Reply:
> then the user changes other fields in the spec, controller sends this to Atlas but no changes happen there as the cluster is paused? The AtlasCluster resource is in "ready: true" state though?

The cluster will be in a non-ready state because such updates will fail on Atlas' side. I'm not sure whether we want to make it "ready" in this scenario though...

Reply:
as discussed p2p having the cluster as "not ready" is a correct outcome 👍  This is something we can confirm with the test